### PR TITLE
Channel availability

### PR DESCRIPTION
**Pull request by d-wysocki**
I want to merge this change because...I want to add channel availability

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/channel/migrations/0002_channel_availability.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+from django.db import migrations, models
+
+
+def set_default_flag(apps, schema):
+    Channel = apps.get_model("channel", "Channel")
+    for channel in Channel.objects.iterator():
+        if not channel.is_active:
+            channel.is_active = True
+        channel.save(update_fields=["is_active"])
```
Comment (tomaszszymanski129) [inline saleor/channel/migrations/0002_channel_availability.py:11]: Why iterate over when same can be achieved on DB level? `Channel.objects.filter(is_active=False).update(is_active=True)`?

Comment (tomaszszymanski129) [reply saleor/channel/migrations/0002_channel_availability.py:11]: Furthermore - is this even needed? in line 21 we add this field and it defaults to True? Please let me know if im missing something.

File: saleor/graphql/checkout/mutations.py [hunk 2]
```diff
@@ -241,20 +241,34 @@ def retrieve_billing_address(cls, user, data: dict) -> Optional[models.Address]:
...
+        except Channel.DoesNotExist:
+            raise ValidationError(
+                {
+                    "channel": ValidationError(
+                        f"Channel with '{channel_slug}' slug does not exist.",
+                        code=CheckoutErrorCode.NOT_FOUND.value,
+                    )
+                }
+            )
+        else:
```
Comment (tomaszszymanski129) [inline saleor/graphql/checkout/mutations.py:256]: redundant else

File: saleor/channel/migrations/0002_channel_availability.py (hunk 1 above)
Comment (d-wysocki) [reply saleor/channel/migrations/0002_channel_availability.py:11]: This data migration is helper for work on feature branch and it should be deleted before merge to master. I will add comment for that.

File: saleor/channel/migrations/0002_channel_availability.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Generated by Django 3.1 on 2020-09-18 07:17
+
+from django.db import migrations, models
+
+
+def set_default_flag(apps, schema):
+    # TODO: delete it before merge to master
```
Comment (tomaszszymanski129) [inline saleor/channel/migrations/0002_channel_availability.py:7]: 👮

**Review by tomaszszymanski129 — APPROVED** [review]

File: saleor/channel/models.py [hunk 4]
```diff
@@ -6,6 +6,7 @@
 
 class Channel(models.Model):
     name = models.CharField(max_length=250)
+    is_active = models.BooleanField(default=True)
```
Comment (fowczarek) [inline saleor/channel/models.py:9]: I'm not sure that we should make the channel as active by default.

File: saleor/channel/utils.py [hunk 5]
```diff
@@ -26,6 +26,10 @@ def get_default_channel() -> Channel:
     try:
         channel = Channel.objects.get()
     except Channel.MultipleObjectsReturned:
+        channels = Channel.objects.filter(is_active=True)
+        if channels.count() == 1:
+            warnings.warn(DEPRECATION_WARNING_MESSAGE)
+            return channels[0]
```
Comment (fowczarek) [inline saleor/channel/utils.py:32]:
```suggestion
        channels = list(Channel.objects.filter(is_active=True))
        if len(channels) == 1:
            warnings.warn(DEPRECATION_WARNING_MESSAGE)
            return channels[0]
```
Your solution generates an additional DB query.

File: saleor/graphql/channel/mutations.py [hunk 6]
```diff
@@ -221,3 +225,71 @@ def clean_channels(cls, info, input, errors: ErrorType, error_code) -> Dict:
...
+
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of the channel to activate.")
+
+    @classmethod
+    def clean_channel_availability(cls, channel):
+        if channel.is_active:
+            raise ValidationError(
+                {
+                    "channel": ValidationError(
```
Comment (fowczarek) [inline saleor/graphql/channel/mutations.py:241]:
```suggestion
                    "id": ValidationError(
```

File: saleor/graphql/channel/mutations.py [hunk 7]
```diff
@@ -221,3 +225,71 @@ def clean_channels(cls, info, input, errors: ErrorType, error_code) -> Dict:
...
+
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of the channel to deactivate.")
+
+    @classmethod
+    def clean_channel_availability(cls, channel):
+        if channel.is_active is False:
+            raise ValidationError(
+                {
+                    "channel": ValidationError(
```
Comment (fowczarek) [inline saleor/graphql/channel/mutations.py:275]:
```suggestion
                    "id": ValidationError(
```

File: saleor/channel/models.py (hunk 4 above)
Comment (maarcingebala) [reply saleor/channel/models.py]: Agree, we can make it inactive, just as we do with newly created products. Once users decide they want to publish the channel they'll do it explicitly.

File: saleor/graphql/schema.graphql [hunk 8]
```diff
@@ -893,6 +907,7 @@ type ChannelUpdate {
 }
 
 input ChannelUpdateInput {
+  isActive: Boolean
```
Comment (maarcingebala) [inline saleor/graphql/schema.graphql:910]: We can manage the `isActive` flag through "update" and "create" mutations, plus we have dedicated mutations for that. Is this on purpose? Do we need them in this case?

Comment (d-wysocki) [reply saleor/graphql/schema.graphql:910]: These changes were made as intended in ticket description.

**Review by maarcingebala — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Pull request merged**